### DISCUSSION TITLE
can't run samples error  in FD4B.Document.pas         "  AddOrUpdate(F.Name, FloatToObj(F.DataType, F.GetValue(self))); "

### DISCUSSION THREAD

**Discussion by drroupen** [SchneiderInfosystems/FB4D#169] [Q&A]
I'm working on Alexandria and tried to compile every project in samples folder but failed I got this error
      in FD4B.Document.pas         "  AddOrUpdate(F.Name, FloatToObj(F.DataType, F.GetValue(self))); " 
any help? thank you in advance
![screenshot](https://github.com/SchneiderInfosystems/FB4D/assets/154215434/87289b36-0ab9-48a0-b430-5a84509f068e)

**Comment by SchneiderInfosystems** [comment]
Can you change this line shortly:
_AddOrUpdate(F.Name, FloatToObj(F.**FieldType**, F.GetValue(self)));_
I have to check this point and correct it after all passed tests.

**Reply by drroupen** [reply]
Yes its ok now, I'll check if this effect elswhere. 
Thank you.

**Reply by SchneiderInfosystems** [reply]
Today, I cannot explain why I did not see this issue when finishing version 1.6. 
It is related to the new "object to document mapper" function SaveObjectToDocument when you declare a member variable of type single, double, or extended in the derived class of TFirestoreDocument.

**Reply by drroupen** [reply]
Good job.
Thank you

**Comment by SchneiderInfosystems** [comment]
Tested and solved within last commit https://github.com/SchneiderInfosystems/FB4D/commit/830c38c74c7fb72485a04eb186cb5c178f46b73e
This problem affects Delphi 11 and older versions, but not Delphi 12.
